src/util/baseutil.js

```diff
@@ -348,3 +348,46 @@ export function isEmpty(value) {
   }
   return true;
 }
+
+/**
+ * Check if IntersectionObserver is supported
```

Comment:
I'm not sure I follow the flow here.. Can you detail how this will be used an add a test that simulates the function calls?

Reply:
when an image component is mounted it will call the detectIntersection function, and pass a 'onIntersect' callback

Reply:
I'll add tests and update

Reply:
@strausr please see the tests i've added and let me know if this can be resolved